x/merkledb/trieview.go

```diff
@@ -692,22 +749,26 @@ func (t *trieView) compressNodePath(parent, node *node) error {
 		)
 		// There is only one child, but we don't know the index.
 		// "Cycle" over the key/values to find the only child.
-		// Note this iteration once because len(node.children) == 1.
-		for index, entry := range node.children {
-			childPath = node.key.AppendExtend(index, entry.compressedPath)
+		// Note this iteration once because len(n.children) == 1.
+		for index, entry := range n.children {
+			childPath = n.key.AppendExtend(index, entry.compressedPath)
 			childEntry = entry
 		}
 
 		nextNode, err := t.getNodeWithID(childEntry.id, childPath, childEntry.hasValue)
 		if err != nil {
 			return err
 		}
-		node = nextNode
+		n = nextNode
 	}
 
 	// [node] is the first node with multiple children.
 	// combine it with the [node] passed in.
-	parent.addChild(node)
+	if parent == nil {
+		t.root = maybe.Some[*node](n)
+		return nil
+	}
+	parent.addChild(n)
```

Comment:
isn't parent still nil here?

Reply:
idk if this comment is outdated now but I don't think `parent` can be `nil` here since we return in the `if` above